client/src/hooks/useFindTrip.tsx

```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from 'react';
+import { TripEntry, getTrip, placeholder } from '../lib/data';
+
+export default function useFindTrip(userId: number, tripId: number): TripEntry {
+  const [trip, setTrip] = useState<TripEntry>(placeholder);
+
+  useEffect(() => {
+    async function readTrips(): Promise<void> {
+      try {
+        const activeTrip = await getTrip(userId, tripId);
+        if (activeTrip.length > 0) setTrip(activeTrip[0]);
+      } catch (error) {
+        console.error('Error getting trips', error);
+      }
+    }
+    readTrips();
+  }, [userId, tripId]);
+
+  if (trip.startDate) trip.startDate = new Date(trip.startDate);
```

Comment:
Doesn't the server send these back as `Date`s? If not, I would do this conversion in `getTrip` instead, so that it's always done regardless of who makes the fetch call.